Assets/MixedRealityToolkit/_Core/Devices/OpenVR/GenericOpenVRController.cs

```diff
@@ -4,13 +4,464 @@
+                        //Get the current Trigger axis state - ** Does not WORK
+                        var triggerAxis = ControllerHandedness == Handedness.Left ? Input.GetAxis(OpenVRInputMappings.OPENVR_TRIGGER_LEFT_CONTROLLER) : Input.GetAxis(OpenVRInputMappings.OPENVR_TRIGGER_RIGHT_CONTROLLER);
+
+                        //Update the interaction data source
+                        interactionMapping.SetFloatValue(triggerAxis);
+
+                        // If our value changed raise it.
+                        if (interactionMapping.Changed)
+                        {
+                            Debug.LogWarning($"Trigger pressed for [{ControllerHandedness}] hand, set to [{triggerAxis}] value");
+
+                            if (triggerAxis > 0)
+                            {
+                                //Raise input system Event if it enabled
+                                InputSystem?.RaiseOnInputPressed(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction, triggerAxis);
+                            }
+                            else
+                            {
+                                //TODO Needs "Released Event"
+                                InputSystem?.RaiseOnInputPressed(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction, triggerAxis);
+                            }
+                        }
+                        break;
+                    }
+                default:
+                    throw new IndexOutOfRangeException();
+            }
+        }
+
+        /// <summary>
+        /// Update the Menu button state.
+        /// </summary>
+        /// <param name="interactionSourceState"></param>
+        /// <param name="interactionMapping"></param>
+        private void UpdateMenuData(MixedRealityInteractionMapping interactionMapping)
+        {
+            //Get the current Menu button state
+            var menuButton = ControllerHandedness == Handedness.Left ? Input.GetKey(KeyCode.JoystickButton2) : Input.GetKey(KeyCode.JoystickButton0);
+            //Update the interaction data source
+            interactionMapping.SetBoolValue(menuButton);
+
+            // If our value changed raise it.
+            if (interactionMapping.Changed)
+            {
+                Debug.LogWarning($"Menu pressed for [{ControllerHandedness}] hand, set to [{menuButton}]");
+
+                //Raise input system Event if it enabled
+                if (menuButton)
+                {
+                    InputSystem?.RaiseOnInputDown(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction);
+                }
+                else
+                {
+                    InputSystem?.RaiseOnInputUp(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction);
+                }
+            }
+        }
+
+        #endregion Update data functions
+
+        #region Nested Types
+        private class OpenVRInputMappings
```

Comment:
Why make a nested class just for const values?

We could probably move this into the main class.

Reply:
I made them a sub class to make them easier to refer to :D

Reply:
Added a remark at the top of the class to state it's purpose.